testing/compare_ids_test.bzl

```diff
@@ -0,0 +1,65 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#Implementation of compare_ids_test
+def _compare_ids_test_impl(ctx):
+    tar_files = []
+    for tar in ctx.attr.tars:
+        tar_files += list(tar.files)
+
+    if (len(tar_files) == 0):
+        fail("No tar files provided for test.")
+
+    if (len(tar_files) == 1 and ctx.attr.id == "0"):
+        fail("One tar provided. Need either second tar or an id to compare it to.")
+
+    tars_string = ""
+    for tar in tar_files:
+        tars_string += tar.path + " "
+
+    runfiles = ctx.runfiles(files = tar_files)
+
+    ctx.actions.expand_template(
```

Comment:
I don't think so, since the test doesn't actually run the script, it just produces an executable which is then run as the test. I think you would need to pass the arguments when you actually use the test, not in its implementation.

Reply:
ack